python/ray/serialization.py

```diff
@@ -184,6 +184,11 @@ def object_id_deserializer(serialized_obj):
                 # UniqueIDs are serialized as
                 # (class name, (unique bytes,)).
                 outer_id = context.get_outer_object_id()
+                # outer_id is None in the case that this ObjectID was closed
+                # over in a function or pickled directly using pickle.dumps().
+                # TODO(edoakes): make sure these objects are pinned forever.
+                if outer_id is None:
```

Comment:
Move this bit to #7358?

Reply:
Ah yeah #7358 will fix it but this prevents a nasty segfault. If the other gets merged first I'll remove this.